target/scripts/helpers/log.sh

```diff
@@ -1,5 +1,103 @@
 #! /bin/bash
 
+LOG_RESET='\e[0m'
+LOG_LGRAY='\e[37m'
+LOG_LBLUE='\e[94m'
+LOG_BLUE='\e[34m'
+LOG_LYELLOW='\e[93m'
+LOG_RED='\e[91m'
+
+# ### DMS Logging Functionality
+#
+# This function provides the logging for scripts used by DMS.
+# It adheres to the convention for log levels.
+# Valid values (in order of increasing verbosity) are: `error`,
+# `warn`, `info`, `debug` and `trace`. The default log level
+# is `info`.
+#
+# #### Arguments
+#
+# $1 :: the log level to log the message with
+# $2 :: the message
+#
+# #### Panics
+#
+# If the first argument is not set or invalid, an error
+# message is logged. Likewise when the second argument
+# is missing. Both failures will return with exit code '1'.
+function _log
+{
+  if [[ -z ${1+set} ]]
```

Comment:
`+` expands if the variable is set. I'm using this construct everywhere I check whether a variable is set, and according to [this SO post](https://stackoverflow.com/questions/3601515/how-to-check-if-a-variable-is-set-in-bash), this is the best way of doing it.

While `:-` would be very much fine as well (for this use case!), there is no need in changing it :)

Reply:
I understand, what `:+` does 😉 I just wondered, what the benefit in returning a string to the test function is, compared to return nothing like we otherwise do. Looks like overhead to me.

PS: `$1` would also be enough, as we don't use `set -u` in any script, that does source this function.

Reply:
Yes, adding `set -u` to `start-mailserver.sh` would be nice! You're right, this must be done carefully (and in another PR of course 😉).

Reply:
I will resolve this because I like the `+` and we can afford 3 more bytes in RAM :D

Reply:
ack